livre/eq/Channel.cpp

```diff
@@ -64,13 +62,63 @@ namespace livre
 const float nearPlane = 0.1f;
 const float farPlane = 15.0f;
 
+struct RedrawFilter : public Filter
+{
+
+public:
+
+    RedrawFilter( Channel* channel )
+        : _channel( channel )
+    {}
+
+    ~RedrawFilter() {}
+    void execute( const FutureMap& input,
+                  PromiseMap& ) const final
+    {
+        waitForAny( input.getFutures( ));
+
+        const bool isRenderingDone =  input.get< bool >( "RenderingDone" )[ 0 ];
+        if( !isRenderingDone )
+            _channel->getConfig()->sendEvent( REDRAW );
+    }
+
+    DataInfos getInputDataInfos() const final
+    {
+        return
+        {
+            { "CacheObjects", getType< ConstCacheObjects >() },
+            { "RenderingDone", getType< bool >() }
+        };
```

Comment:
That's an initializer list taking tuples of strings and typeid(?). Look at the variadic template constructor that understands that if you want to spend an hour until you fully understand all the pieces involved.